framework/src/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -2411,18 +2432,46 @@ protected void addAliasedAnnotation(String canonicalName, AnnotationMirror type)
     }
 
     /**
+     * Add the annotation whose canonical name is given by {@code canonicalName} as an alias for the
+     * annotation {@code type}. An {@code copyElements} flag is passed in to indicate whether we
+     * want to copy the elements over when getting the alias from the canonical one. An optional
+     * {@code ignorableElements} contains a list of elements that can be safely dropped when the
+     * elements are being copied over.
+     */
+    protected void addAliasedAnnotation(
+            String canonicalName,
+            AnnotationMirror type,
+            boolean copyElements,
+            String... ignorableElements) {
+        addAliasedAnnotation(canonicalName, type);
+        aliasesCopyElements.put(canonicalName, copyElements);
+        aliasesIgnorableElements.put(canonicalName, ignorableElements);
+    }
+
+    /**
      * Returns the canonical annotation for the passed annotation if it is an alias of a canonical
      * one in the framework. If it is not an alias, the method returns null.
      *
-     * <p>Returns an aliased type of the current one
+     * <p>Returns an aliased type of the current one. The attributes contained by the passed
+     * annotation are not copied over unless specified when {@link #addAliasedAnnotation(String,
+     * AnnotationMirror, boolean, String...)}
      *
      * @param a the qualifier to check for an alias
      * @return the alias or null if none exists
      */
     public /*@Nullable*/ AnnotationMirror aliasedAnnotation(AnnotationMirror a) {
         TypeElement elem = (TypeElement) a.getAnnotationType().asElement();
         String qualName = elem.getQualifiedName().toString();
-        return aliases.get(qualName);
+        AnnotationMirror frameworkAnno = aliases.get(qualName);
+        if (frameworkAnno != null && a.getElementValues().size() > 0) {
+            AnnotationBuilder builder = new AnnotationBuilder(processingEnv, frameworkAnno);
+            if (aliasesCopyElements.getOrDefault(qualName, false)) {
```

Comment:
As you only care whether or not to copy elements, you could make `aliasesCopyElements` a `Set` and check for containment here. That should be a bit simpler to check and store.